### PR TITLE
Fix/document id validation

### PR DESCRIPTION
**Pull request by achocron**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description
Fail fast if document ID contains ?, \ , /, #, or whitespace.

### CODE REVIEW

**Comment by achocron** [comment]
Not sure what the submodule changes are. Considering reverting those, as they just appeared without any action from me.

**Comment by jaelim-ms** [comment]
MSAL project is configured to use repo root for their temporary build files, it will contain all new files if you just stage all files in your App Center SDK root. I don't know why they don't have the `build` folder in `.gitignore`, it should be changed in MSAL. In the meantime, please make sure you do not stage MSAL project.

File: AppCenterData/AppCenterData/MSData.m [hunk 1]
```diff
@@ -638,6 +652,23 @@ - (MSDataError *)generateInvalidClassError {
...
+  NSRegularExpression *expr = [NSRegularExpression regularExpressionWithPattern:kMSDocumentIdValidationPattern options:0 error:nil];
+  NSUInteger matchCount = [expr numberOfMatchesInString:documentId options:0 range:NSMakeRange(0, documentId.length)];
+  return matchCount == 0;
+}
+
+- (MSDataError *)generateInvalidDocumentIdError {
+  MSDataError *dataError = [[MSDataError alloc] initWithErrorCode:MSACDataErrorDocumentIdInvalid
+                                                       innerError:nil
+                                                          message:(NSString *)kMSACDataErrorDocumentIdInvalidDesc];
+  MSLogError([MSData logTag], @"Document ID is invalid.");
```
Comment (Jamminroot) [inline AppCenterData/AppCenterData/MSData.m]: I think we can use kMSACDataErrorDocumentIdInvalidDesc here, too

File: AppCenterData/AppCenterData/MSData.m [hunk 2]
```diff
@@ -638,6 +652,23 @@ - (MSDataError *)generateInvalidClassError {
...
+  return dataError;
```
Comment (Jamminroot) [inline AppCenterData/AppCenterData/MSData.m]: Is it ok that declared return type does not match actually returned value type? I'd prefer explicit conversion over implicit

Comment (achocron) [reply AppCenterData/AppCenterData/MSData.m]: Not sure what you mean since they are both `MSDataError`

**Pull request merged**